addon_service/oauth1/models.py

```diff
@@ -0,0 +1,33 @@
+from django.db import models
+
+from addon_service.common.base_model import AddonsServiceBaseModel
+
+
+class OAuth1ClientConfig(AddonsServiceBaseModel):
+    """
+    Model for storing attributes that are required for managing
+    OAuth1 credentials exchanges with an ExternalService on behalf
+    of a registered client (e.g. the OSF)
+    """
+
+    # URI that allows to obtain temporary request token to proceed with user auth
+    request_token_url = models.URLField(null=False)
+    # URI to which user will be redirected to authenticate
+    auth_url = models.URLField(null=False)
+    # URI to which user will be redirected after authentication
+    auth_callback_url = models.URLField(null=False)
+    # URI to obtain access token
+    access_token_url = models.URLField(null=False)
+
+    client_key = models.CharField(null=True)
+    client_secret = models.CharField(null=True)
```

Comment:
hmm, the client secret should be stored encrypted -- but since OAuth2ClientConfig is the same, maybe worth splitting that into another ticket? (...and i'm starting to reconsider the `EncryptedDataclassModel` abstract base, now that we have three potential uses for it...)

Reply:
I agree, storing any unencrypted credentials in the db, poses security risks

Reply:
Yes, separate ticket. Need input from cloud eng/devops as to whether they want this in the database at all or if they'd prefer an approach like [this one](https://github.com/CenterForOpenScience/gravyvalet/pull/77)